pkg/ingester/shipper_test.go

```diff
@@ -329,3 +336,137 @@ func TestReadThanosMetaFile(t *testing.T) {
+					},
+				},
+			},
+			oooCompactionHintExpected: false,
+			expectedLabels:            map[string]string{},
+		},
+		{
+			name:        "in-order block, addOOOLabel = true",
+			addOOOLabel: false,
+			meta: metadata.Meta{
+				BlockMeta: tsdb.BlockMeta{
+					ULID:    ulid.MustNew(1, nil),
+					MaxTime: 2000,
+					MinTime: 1000,
+					Version: 1,
+					Stats: tsdb.BlockStats{
+						NumSamples: 100, // Shipper checks if number of samples is greater than 0.
+					},
+				},
+			},
+			oooCompactionHintExpected: false,
+			expectedLabels:            map[string]string{},
+		},
+		{
+			name:        "OOO block, addOOOLabel = false",
+			addOOOLabel: false,
+			meta: metaWithOOOHint(metadata.Meta{
+				BlockMeta: tsdb.BlockMeta{
+					ULID:    ulid.MustNew(1, nil),
+					MaxTime: 2000,
+					MinTime: 1000,
+					Version: 1,
+					Stats: tsdb.BlockStats{
+						NumSamples: 100, // Shipper checks if number of samples is greater than 0.
+					},
+				},
+			}),
+			oooCompactionHintExpected: true,
+			expectedLabels:            map[string]string{},
+		},
+		{
+			name:        "OOO block, addOOOLabel = true",
+			addOOOLabel: true,
+			meta: metaWithOOOHint(metadata.Meta{
+				BlockMeta: tsdb.BlockMeta{
+					ULID:    ulid.MustNew(1, nil),
+					MaxTime: 2000,
+					MinTime: 1000,
+					Version: 1,
+					Stats: tsdb.BlockStats{
+						NumSamples: 100, // Shipper checks if number of samples is greater than 0.
+					},
+				},
+			}),
+			oooCompactionHintExpected: true,
+			expectedLabels:            map[string]string{mimir_tsdb.OutOfOrderExternalLabelKey: OutOfOrderExternalLabelValue},
+		},
+		{
+			name:        "OOO block, addOOOLabel = true, additional labels",
+			addOOOLabel: true,
+			meta: metaWithOOOHint(metadata.Meta{
+				BlockMeta: tsdb.BlockMeta{
+					ULID:    ulid.MustNew(1, nil),
+					MaxTime: 2000,
+					MinTime: 1000,
+					Version: 1,
+					Stats: tsdb.BlockStats{
+						NumSamples: 100, // Shipper checks if number of samples is greater than 0.
+					},
+				},
+				Thanos: metadata.Thanos{Labels: map[string]string{"a": "b"}},
+			}),
+			oooCompactionHintExpected: true,
+			expectedLabels:            map[string]string{"a": "b", mimir_tsdb.OutOfOrderExternalLabelKey: OutOfOrderExternalLabelValue},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			blocksDir := t.TempDir()
+			bucketDir := t.TempDir()
+
+			bkt, err := filesystem.NewBucketClient(filesystem.Config{Directory: bucketDir})
+			require.NoError(t, err)
+
+			logs := &concurrency.SyncBuffer{}
+			logger := log.NewLogfmtLogger(logs)
+			tenantLimits := map[string]*validation.Limits{
+				"": {
+					OutOfOrderBlocksExternalLabelEnabled: tc.addOOOLabel,
+				},
+			}
+			overrides, err := validation.NewOverrides(defaultLimitsTestConfig(), validation.NewMockTenantLimits(tenantLimits))
+			require.NoError(t, err)
+			s := NewShipper(logger, overrides, "", nil, blocksDir, bkt, metadata.TestSource)
```

Comment:
[nit] We don't need fullblown overrides here. Struct (or even just bool) with single `OutOfOrderBlocksExternalLabelEnabled` method would be enough.